apps/pollbook/backend/src/app_multi_node.test.ts

```diff
@@ -536,6 +545,514 @@ test('connection status is managed properly with many pollbooks', async () => {
+    );
+    vitest.advanceTimersByTime(NETWORK_POLLING_INTERVAL);
+    for (const context of [pollbookContext1, pollbookContext2]) {
+      context.mockUsbDrive.insertUsbDrive({});
+      expect(await context.localApiClient.getDeviceStatuses()).toMatchObject({
+        network: {
+          isOnline: false,
+          pollbooks: [],
+        },
+      });
+    }
+    // Mock hasOnlineInterface to always return true
+    vi.mocked(hasOnlineInterface).mockResolvedValue(true);
+    const { port: port1 } =
+      pollbookContext1.peerServer.address() as AddressInfo;
+    const { port: port2 } =
+      pollbookContext2.peerServer.address() as AddressInfo;
+
+    vi.spyOn(
+      pollbookContext1.peerWorkspace.store,
+      'getNewEvents'
+    ).mockResolvedValue({
+      events: [],
+      hasMore: false,
+    });
+    vi.spyOn(
+      pollbookContext2.peerWorkspace.store,
+      'getNewEvents'
+    ).mockResolvedValue({
+      events: [],
+      hasMore: false,
+    });
+
+    vi.spyOn(AvahiService, 'discoverHttpServices').mockResolvedValue([
+      {
+        name: 'Pollbook-test-0',
+        host: 'local',
+        resolvedIp: 'localhost',
+        port: port1.toString(),
+      },
+      {
+        name: 'Pollbook-test-1',
+        host: 'local',
+        resolvedIp: 'localhost',
+        port: port2.toString(),
+      },
+    ]);
+    vitest.advanceTimersByTime(NETWORK_POLLING_INTERVAL);
+
+    await extendedWaitFor(async () => {
+      expect(
+        await pollbookContext1.localApiClient.getDeviceStatuses()
+      ).toMatchObject({
+        network: {
+          isOnline: true,
+          pollbooks: [
+            expect.objectContaining({
+              status: PollbookConnectionStatus.MismatchedConfiguration,
+            }),
+          ],
+        },
+      });
+    });
+
+    vitest.advanceTimersByTime(NETWORK_POLLING_INTERVAL);
+
+    await extendedWaitFor(async () => {
+      expect(
+        await pollbookContext2.localApiClient.getDeviceStatuses()
+      ).toMatchObject({
+        network: {
+          isOnline: true,
+          pollbooks: [
+            expect.objectContaining({
+              status: PollbookConnectionStatus.MismatchedConfiguration,
+            }),
+          ],
+        },
+      });
+    });
+
+    // Before connecting the pollbooks should not have queried one another for events
+    expect(
+      pollbookContext1.peerWorkspace.store.getNewEvents
+    ).not.toHaveBeenCalled();
+    expect(
+      pollbookContext2.peerWorkspace.store.getNewEvents
+    ).not.toHaveBeenCalled();
+
+    // Set the election without setting configured precinct should not connect
+    pollbookContext1.workspace.store.setElectionAndVoters(
+      electionDefinition,
+      'fake-package-hash',
+      testStreets,
+      testVoters
+    );
+    pollbookContext2.workspace.store.setElectionAndVoters(
+      electionDefinition,
+      'fake-package-hash',
+      testStreets,
+      testVoters
+    );
```

Comment:
Seems like there's some repeated boilerplate between these tests. Can we factor out what is convenient to helper functions?